src/frontend/test_runner/tests/testdata/agg.yaml

```diff
@@ -496,3 +496,42 @@
         StreamExchange { dist: Single }
           StreamLocalSimpleAgg { aggs: [count, sum($0) filter(($0 < 5:Int32))] }
             StreamTableScan { table: t, columns: [v2, _row_id], pk_indices: [1] }
+- sql: |
+    /* distinct agg and non-disintct agg */
+    create table t(a int, b int, c int);
+    select a, count(distinct b) as distinct_b_num, sum(c) as sum_c from t group by a;
+  optimized_logical_plan: |
+    LogicalAgg { group_key: [0], agg_calls: [sum($2) filter(($1 = 0:Int64)), sum($3) filter(($1 = 1:Int64))] }
+      LogicalProject { exprs: [$0, $2, $3, $4] }
+        LogicalAgg { group_key: [0, 1, 3], agg_calls: [count($1), sum($2)] }
+          LogicalExpand { column_subsets: [[0, 1], [0, 2]] }
+            LogicalScan { table: t, columns: [a, b, c] }
```

Comment:
This plan looks problematic:
Assume `t` has the following rows:
```
1 2 3
1 2 4
```
The expected output is:
```
1 1 7
```

Without the ability to actually run it e2e, let's execute it in mind:
After expand:
```
1 2 N 0
1 2 N 0
1 N 3 1
1 N 4 1
```
Then `LogicalAgg { group_key: [0, 1, 3], agg_calls: [count($1), sum($2)] }`:
```
1 2 0 2 N
1 N 1 0 7
```
Then project:
```
1 0 2 N
1 1 0 7
```
Finally:
```
1 2 7
```

The error **in this case** is that we `count` when group by `a, b, flag`, then `sum` that result. However, The 1st `count` is **unnecessary**, we just need to `group by a, b, flag` and then `count` the `b` from group column. This also aligns with the original comment in issue.

Intuitively, for an agg-distinct call, the earlier agg node does `distinct`, and the later agg node does the real `agg`. At the same time, for an agg without distinct, the earlier agg node does the real `agg`, and the later agg node may be able to use `single_value` (correct me if I am wrong).

---

The "fix" above cannot handle the case `count(distinct b) filter (where d > 5)`, where `d` is neither a group column or used by any other agg calls. We will have to `filter d > 5` before we can `distinct b` via `group by b` of the earlier agg node. Some premature ideas I just come up with:
* `count(*) filter (where d > 5)` in earlier agg node, then `filter(where flag = x and count_star > 0)`. Note that it is always `count` no matter what the original agg call is. This feels like a bit wasteful.
* add a filter node between expand and earlier agg node